config/7.0/obdemo-bank/ig/scripts/groovy/ProcessDetachedSig.groovy

```diff
@@ -0,0 +1,56 @@
+import org.forgerock.http.protocol.*
+import org.forgerock.json.jose.utils.Utils
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import org.forgerock.json.jose.*
+import org.forgerock.json.jose.common.JwtReconstruction
+import org.forgerock.json.jose.jws.SignedJwt
+import org.forgerock.json.jose.jwk.store.JwksStore.*
+import org.forgerock.secrets.jwkset.JwkSetSecretStore
+import org.forgerock.json.JsonValueFunctions.*
+import org.forgerock.json.jose.jwk.JWKSetParser
+import org.forgerock.util.Options
+import org.forgerock.util.time.Duration
+import java.net.URL
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.Future
+
+// On the way in, build a JWT from the header and content to pass to the JWT verification filter
+
+logger.debug("Building JWT from detached header")
+def header = request.headers.get(headerName)
```

Comment:
I don't know how this voodoo works exactly, but it looks like headerName is resolved from the ```"headerName": "x-jws-signature",``` above.

Reply:
headerName is passed in as an argument in the route that calls this script - i.e. this from `12-ob-payment-submission.json`

```        {
          "name": "ProcessDetachedSig",
          "type": "ScriptableFilter",
          "config": {
            "type": "application/x-groovy",
            "file": "ProcessDetachedSig.groovy",
            "clientHandler": "IDMClientHandler",
            "args": {
              "headerName": "x-jws-signature",
              "idmBaseUri": "https://&{iam.fqdn}",
              "objTpp": "obTpp",
              "secretStore": "${heap['JwkSetSecretStore-ClientJwks']}"
            }
          }
        }```

Maybe we could have a convention for variable names when they are route arguments, so that it's clear where they come from.

Reply:
Looking at this, I should really tidy up the script arguments - some of these are from earlier iterations when we talked to IDM and manipulated the secret store.